stellargraph/data/explorer.py

```diff
@@ -414,30 +414,14 @@ def run(
         if weighted:
             self._check_weights_valid()
 
-            # calculate the appropriate unnormalised transition probability, given the history of
-            # the walk
-            def transition_probability(nn):
-                nn, weight = nn
-
-                if nn == previous_node:  # d_tx = 0
-                    return ip * weight
-                elif any(nn == pn for pn, _ in previous_node_neighbours):  # d_tx = 1
-                    return weight
-                else:  # d_tx = 2
-                    return iq * weight
+        cast_func = np.cast[self.graph._edges.weights.dtype]
+        ip = cast_func(1.0 / p)
+        iq = cast_func(1.0 / q)
 
-        else:
-            # without weights
-            def transition_probability(nn):
-                if nn == previous_node:  ## d_tx = 0
-                    return ip
-                elif nn in previous_node_neighbours:  # d_tx = 1
-                    return 1.0
-                else:  # d_tx = 2
-                    return iq
-
-        ip = 1.0 / p
-        iq = 1.0 / q
+        if np.isinf(ip):
+            raise ZeroDivisionError("p: value is too small")
+        if np.isinf(iq):
+            raise ZeroDivisionError("q: value is too small")
```

Comment:
I don't think this is necessary: the `1.0 / p` division still happens in Python-space, and so still throws an error. If it was `cast_func(1.0) / cast_func(p)` that might be different, but that's not the case.

```suggestion
```

Reply:
I guess this `ZeroDivisionError` is slightly misleading. When `p` is a really small float `ip` is cast to `np.inf` e.g. 

```
q = 1e-100
iq = 1 / q
print(np.float32(iq)) # np.inf here
```

Maybe another error would more appropriate here because its not _technically_ a zero division error.

Reply:
That sounds like a good approach, but we could instead phrase it as something like "1/p should be representable by the data type fo the edge weights", and not hard code a bound.

Reply:
@kieranricardo did you miss applying something along the lines of @kjun9's suggestion before re-requesting a review?